File: 35-SearchInsertPosition/35-SearchInsertPosition.c
// Last updated: 9/14/2025, 9:18:35 AM
int searchInsert(int* nums, int numsSize, int target){
          int start=0,end=numsSize-1;
          if(nums[0]>target){
              return start;
          }else if(nums[numsSize-1]<target){
               return numsSize;
          }
          while(start<=end){
              int mid=start+(end-start)/2;
              if(nums[mid]==target){
                  return mid;
              }else if(nums[mid]>target){
                  end=mid-1;
              }else{
                  start=mid+1;
              }
          }
          return start;
}

File: 35-SearchInsertPosition/35-SearchInsertPosition.java
// Last updated: 9/14/2025, 9:18:31 AM
class Solution {
    public int searchInsert(int[] nums, int target) {
        //lower bound approach
        int ans = nums.length;
        int start = 0;
        int end = nums.length-1;

        while(start<=end){
            int mid = start + (end - start)/2;
            if(nums[mid]>=target){
                ans = mid;
                end = mid-1;
            }else{
                start = mid+1;
            }
        }
        return ans;

    }
}